src/ReactHardwareComponent.js

```diff
@@ -42,16 +64,23 @@ type ViewConfigPropTypes = {
  * @constructor ReactHardwareComponent
  * @extends ReactComponent
  * @extends ReactMultiChild
- * @param {!object} viewConfig View Configuration.
  */
-const ReactHardwareComponent = function(
-  element: React$Element
-) {
-  this.viewConfig = DEFAULT_VIEW_CONFIG;
+const ReactHardwareComponent = function(element: React$Element) {
   this._rootNodeID = null;
   this._renderedChildren = null;
   this._io = null;
-  this.construct(element);
+  this._currentElement = element;
+  this.viewConfig = DEFAULT_VIEW_CONFIG;
+  if (typeof element.type === 'string') {
+    this.viewConfig = getViewConfig(element.type);
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    invariant(
```

Comment:
ReactDOM when given something like `<foobar />` can still create and render it. Do you happen to know what RN does off hand?

Currently I’d rather change this to a `warning` and just revert the type to a `pin`/use the `PIN_VIEW_CONFIG` with `UnknownComponent` as the `uiViewClassName` to not throw an error in the rendering.